enterprise/models.py

```diff
@@ -2870,6 +2889,69 @@ def get_context(self):
 
         return super().get_context()
 
+    @classmethod
+    def get_distinct_assignments_by_role_name(cls, user, role_names=None):
+        """
+        Returns a mapping of role names to sets of enterprise customer uuids
+        for which the user is assigned that role.
+        """
+        # super().get_assignments() returns pairs of (role name, contexts), where
+        # contexts is a list of 1 or more enterprise uuids (or the ALL_ACCESS_CONTEXT token)
+        assigned_customers_by_role = collections.defaultdict(set)
+        for role_name, customer_uuids in super().get_assignments(user, role_names):
+            assigned_customers_by_role[role_name].update(customer_uuids)
+        return assigned_customers_by_role
+
+    @classmethod
+    def get_assignments(cls, user, role_names=None):
+        """
+        Return an iterator of (rolename, enterprise customer uuid) for the given
+        user (and maybe role_names).
+
+        Differs from super().get_assignments(...) in that it yields the (role name, customer uuid)
+        pairs that correspond to the currently *active* EnterpriseCustomerUser for the given role first.
+        The resulting generated pairs are sorted by role name, and within role_name, by (active?, customer uuid).
+
+        ('enterprise_admin', ['active-enterpirse-uuid', 'other-active-enterpirse-uuid', 'inactive-enterpirse-uuid'])
+        ('enterprise_learner', ['active-enterpirse-uuid', 'inactive-enterpirse-uuid']),
+        ('enterprise_openedx_operator', ['*'])
```

Comment:
Sorry, this is inconsistent in terms of how it's talking about the yielded items - I'll clean it up.